File: 17-monitoring/npm-package/monitoring.ts
import axios from 'axios';
const API_BASE = 'https://api.costkatana.com/api'\;
const API_KEY = process.env.COST_KATANA_API_KEY;

async function getMetrics() {
  const response = await axios.get(
    `${API_BASE}/monitoring/metrics`,
    { headers: { 'Authorization': `Bearer ${API_KEY}` } }
  );
  console.log('Metrics:', response.data);
  return response.data;
}

async function getHealth() {
  const response = await axios.get(
    `${API_BASE}/monitoring/health`,
    { headers: { 'Authorization': `Bearer ${API_KEY}` } }
  );
  console.log('Health:', response.data);
  return response.data;
}

export { getMetrics, getHealth };

File: 17-monitoring/python-sdk/monitoring.py
import os, requests
API_BASE = "https://api.costkatana.com/api"
API_KEY = os.getenv("COST_KATANA_API_KEY")

def get_metrics():
    response = requests.get(
        f"{API_BASE}/monitoring/metrics",
        headers={"Authorization": f"Bearer {API_KEY}"}
    )
    response.raise_for_status()
    print("Metrics:", response.json())
    return response.json()

def get_health():
    response = requests.get(
        f"{API_BASE}/monitoring/health",
        headers={"Authorization": f"Bearer {API_KEY}"}
    )
    response.raise_for_status()
    print("Health:", response.json())
    return response.json()
